alicloud/data_source_alicloud_db_instances_sql_log.go

```diff
@@ -0,0 +1,183 @@
+				Optional: true,
+			},
+			"query_key_words": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"data_base": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"user": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"page_size": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				ValidateFunc: validation.IntBetween(30, 100),
+				Default:      30,
+			},
+			"page_number": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  1,
+			},
+			"output_file": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			// Computed values
+			"total_record_count": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"page_record_number": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"page_record_count": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"sql_record": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"db_name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"account_name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"host_address": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"sql_text": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"total_execution_times": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"return_row_counts": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"thread_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"execute_time": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceAlicloudDBInstancesSQLLogRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	request := rds.CreateDescribeSQLLogRecordsRequest()
+
+	request.RegionId = client.RegionId
+	request.DBInstanceId = d.Get("db_instance_id").(string)
+	request.StartTime = d.Get("start_time").(string)
+	request.EndTime = d.Get("end_time").(string)
+	request.Form = "Stream"
+	if v, ok := d.GetOk("sql_id"); ok && v.(string) != "" {
+		request.SQLId = requests.NewInteger(d.Get("sql_id").(int))
+	}
+	if v, ok := d.GetOk("query_key_words"); ok && v.(string) != "" {
+		request.QueryKeywords = d.Get("query_key_words").(string)
+	}
+	if v, ok := d.GetOk("data_base"); ok && v.(string) != "" {
+		request.Database = d.Get("data_base").(string)
+	}
+	if v, ok := d.GetOk("user"); ok && v.(string) != "" {
+		request.User = d.Get("user").(string)
+	}
+	if v, ok := d.GetOk("page_size"); ok && v.(int) != 0 {
+		request.PageSize = requests.NewInteger(d.Get("page_size").(int))
+	}
+	if v, ok := d.GetOk("page_number"); ok && v.(int) != 0 {
+		request.PageNumber = requests.NewInteger(d.Get("page_number").(int))
```

Comment:
this is set by terraform, please refer to data_source_alicloud_cen_instances

Reply:
> this is set by terraform, please refer to data_source_alicloud_cen_instances

Don't have to change